src/SourceBuild/content/eng/init-source-only.proj

```diff
@@ -13,48 +13,64 @@
+          Inputs="$(ExternalTarballsDir)"
+          Outputs="$(PrebuiltSourceBuiltPackagesPath)">
     <MakeDir Directories="$(PrebuiltSourceBuiltPackagesPath)" Condition="'$(CustomPrebuiltSourceBuiltPackagesPath)' == ''" />
     <Exec Command="tar -xzf $(ExternalTarballsDir)$(SourceBuiltArtifactsTarballName).*$(ArchiveExtension)"
-          WorkingDirectory="$(PrebuiltSourceBuiltPackagesPath)"
-          Condition="'$(CustomPrebuiltSourceBuiltPackagesPath)' == ''" />
+          WorkingDirectory="$(PrebuiltSourceBuiltPackagesPath)" />
+  </Target>
 
-    <!--
-      Check for a prebuilt dependency tarball and extract if exists. If there isn't one, we expect
-      the build to be working without prebuilts.
-    -->
+  <!-- Check for a prebuilt dependency tarball and extract if exists. If there isn't one, we expect
+       the build to be working without prebuilts. -->
+  <Target Name="UnpackSourceBuiltPrebuiltsArchive"
+          Inputs="$(ExternalTarballsDir)"
+          Outputs="$(PrebuiltPackagesPath)">
     <ItemGroup>
       <SourceBuiltPrebuiltsTarballFile Include="$(ExternalTarballsDir)$(SourceBuiltPrebuiltsTarballName).*$(ArchiveExtension)" />
     </ItemGroup>
+
     <Exec Command="tar -xzf %(SourceBuiltPrebuiltsTarballFile.FullPath)"
           WorkingDirectory="$(PrebuiltPackagesPath)"
           Condition="'@(SourceBuiltPrebuiltsTarballFile)' != ''" />
+  </Target>
 
-    <!-- Copy SBRP packages to reference packages location -->
+  <!-- Copy SBRP packages to reference packages location. -->
+  <Target Name="CopySourceBuiltReferencePackages"
+          DependsOnTargets="UnpackSourceBuiltArtifactsArchive"
+          Inputs="$(PrebuiltSourceBuiltPackagesPath)SourceBuildReferencePackages"
+          Outputs="$(ReferencePackagesDir)">
     <ItemGroup>
-      <UnpackedSourceBuildReferencePackages Include="$(PrebuiltSourceBuiltPackagesPath)SourceBuildReferencePackages/*"/>
+      <UnpackedSourceBuildReferencePackage Include="$(PrebuiltSourceBuiltPackagesPath)SourceBuildReferencePackages/*" />
     </ItemGroup>
 
-    <Move SourceFiles="@(UnpackedSourceBuildReferencePackages)" DestinationFiles="$(ReferencePackagesDir)%(Filename)%(Extension)" />
-
-    <MakeDir Directories="$(BaseIntermediateOutputPath)" />
-    <Touch Files="$(BaseIntermediateOutputPath)UnpackTarballs.complete" AlwaysCreate="true">
-      <Output TaskParameter="TouchedFiles" ItemName="FileWrites" />
-    </Touch>
+    <!-- Either move or copy the unpacked SBRP packages.
+         Don't move when the packages directory is externally provided. -->
+    <Move SourceFiles="@(UnpackedSourceBuildReferencePackage)"
+          DestinationFolder="$(ReferencePackagesDir)"
+          Condition="'$(CustomPrebuiltSourceBuiltPackagesPath)' == ''" />
+    <Copy SourceFiles="@(UnpackedSourceBuildReferencePackage)"
+          DestinationFolder="$(ReferencePackagesDir)"
+          Condition="'$(CustomPrebuiltSourceBuiltPackagesPath)' != ''" />
```

Comment:
Does this change also fix https://github.com/dotnet/source-build/issues/4325?

Reply:
Yes :)